### DISCUSSION TITLE
Xamarin.Forms Projects migration to MAUI - Timeline

### DISCUSSION THREAD

**Discussion by chriswoodward** [dotnet/maui#242] [General] [answered]
Hi, firstly apologies if this question has already been answered elsewhere. We have several projects written in Xamarin Forms that are Live in the App Store and also on Google Play. We build our projects in Visual Studio 2019 on a PC and, for Xamarin.iOS, we connect to a Mac which then does the build in Xcode. Currently, I believe that when Apple roll out an update to Xcode on the Mac there is a related update actioned by Microsoft to Xamarin.IOS which then becomes available when Visual Studio is updated. So my question is, at what point is Xamarin.iOS going to stop being updated so that it then becomes "out of sync" with Xcode which, as I understand it, will mean that we will no longer be able to build a Xamarin.IOS project. This will then mean we will not be able to do any updates to any of our Apps on the App Store. I believe Microsoft have said that Xamarin.iOS will not be updated as from Nov 2021 so does this mean that potentially it will start to get out of sync with Xcode as from that date ?

**Comment by mrwcjoughin** [comment]
Hi Chris,

You are mistaken.

Xamarin.iOS will not not be supported once MAUI is out, as it effectively replaces that (for all platforms currently targeted by Xamarin)
However I do remember they said Xamarin will still be supported (i.e. bug fixes which would include ensuring remaining compatible with XCode updates) for 2 years after MAUI is out - so therefore you will have 2 years to migrate your apps to MAUI...

**Comment by chriswoodward** [comment]
Thanks for your Reply Matthew though we are not sure this is the case to be honest.

I mentioned this to my boss the other day and we are a bit concerned that there will be insufficient time to migrate our Xamarin.Forms over to MAUI if the cut-off point is November 2021. Therefore, we would be grateful if anyone on the .NET MAUI Team would be good enough to provide some clarification on the migration timelines being proposed.

@davidortinau
@PureWeen

**Reply by PureWeen** [reply]
Xamarin.Forms will be supported with bug fixes until November 2023 (2 years after MAUI releases)

If you aren't using any custom renderers your app should shift over to MAUI with minimal effort.  At this point all the breaking changes we planned for are in 5.0

If you have custom renderers you may need to adapt those slightly but nothing that crazy.

**Reply by davidortinau** [reply]
A little more clarification, the SDK bindings (Xamarin.iOS) will absolutely continue. .NET MAUI depends on it for targeting iOS.  

Your iOS and Android apps will continue to receive updates, and we are making sure you have a smooth upgrade to .NET MAUI.

Specifically regarding Xcode updates: Xcode 12 just shipped and we have iOS for that, and we are already working on the next Xcode (12.2 beta).  

When NET 6 ships previews, we'll start previewing iOS bindings that run on that version of .NET. It'll just be referred to as ".NET for iOS" in the unified .NET 6 world instead of Xamarin.iOS.  

When Apple ships Xcode 13, we'll once again do SDK bindings, and so on and so forth into the future to make sure you can deploy to iOS.

**Reply by EmilAlipiev** [reply]
@davidortinau  @PureWeen How about shell applications? Is Maui XF shell compatible or requires more effort? For example, if i have to start a project for Xamarin.forms, should i start with XF classic or with Shell?

**Comment by PureWeen** [comment, marked as answer]
Xamarin.Forms will be supported with bug fixes until November 2023 (2 years after MAUI releases)

If you aren't using any custom renderers your app should shift over to MAUI with minimal effort. 

At this point all the breaking changes we planned for are in XF 5.0

If you have custom renderers you may need to adapt those slightly but nothing that crazy.

**Reply by radioactiveman** [reply]
@PureWeen: 

The official [blog post](https://devblogs.microsoft.com/dotnet/introducing-net-multi-platform-app-ui/) about MAUI says the final release of Xamarin.Forms will be in November 2022 (one year after the first MAUI release). But you say it will be supported for two years. Could you please clarify this?

> Xamarin.Forms will ship a new major version later this year, and continue to ship minor and service releases every 6 weeks through .NET 6 GA in November 2021. The final release of Xamarin.Forms will be serviced for a year after shipping, and all modern work will shift to .NET MAUI.

**Reply by radioactiveman** [reply]
@PureWeen, @davidortinau: Could you please clarify my question above? Thanks in advance.

**Reply by divyesh008** [reply]
@PureWeen Yes the official blog, even @jamesmontemagno has stated that the Xamarin.Forms will be supported till Nov 2022. But you said it will be for 2 years. Is it correct?

**Reply by jamesmontemagno** [reply]
It is 1 year after release for Xamarin.Forms. Xamarin.iOS and Xamarin.Android have a 2 year policy https://dotnet.microsoft.com/platform/support/policy/xamarin

Xamarin.Forms projects can all be upgraded to .NET MAUI 
Xamarin.iOS, and Xamarin.Andriod can be updated to .NET 6

**Reply by flimtix** [reply]
Since the [latest changes](https://github.com/dotnet/maui/wiki/roadmap), the release of MAUI is delayed to **Q2 2022**. How will this look now with Xamarin support?
